document.py

```diff
@@ -41,9 +45,20 @@ def tokenize_doc(self):
 
     def clean_tokens(self, common_words):
         for attr in Document.tokenized_fields:
-            for token in getattr(self, attr):
-                if token in common_words or token in Document.signs_to_remove:
-                    getattr(self, attr).remove(token)
+            new_attr_value = [
+                token.lower()
+                for token in getattr(self, attr)
+                if not (
+                    token.lower() in common_words or
+                    token.lower() in Document.signs_to_remove
+                )
+            ]
+            setattr(self, attr, new_attr_value)
```

Comment:
I propose here an inline build for the list because I had issues with previous method. When we iterate over the token list, we remove a token from this list which will not apply the remove action. Maybe there is a better undestandable way to write it but I think this is the most efficient. What should we do here? :)